src/main/java/com/artipie/Pie.java

```diff
@@ -95,12 +99,32 @@ public Response response(final String line, final Iterable<Map.Entry<String, Str
         return new AsyncSlice(
             CompletableFuture.supplyAsync(
                 () -> new Unchecked<>(this.settings::storage).value()
-            )
-                .thenComposeAsync(
-                    storage -> storage.value(new Key.From(String.format("%s.yaml", repo)))
+            ).thenComposeAsync(
```

Comment:
@HDouss you are creating an instance of `CompletableFuture<Single<Content>>`, are you sure that exactly this type should be returned here? Mos probably you can use
```java
thenCompose(storage -> storage.value(new Key.From(String.format("%s.yaml", repo))))
```

Reply:
@g4s8 I did not want to touch the stages chaining logic. It was CompletionStage<Content>, and now it's  CompletionStage<Single<Content>>  because we are using publishers (see comments and discussion). The idea to use a publisher was to be able to catch the exception in the final CompletionStage, instead of the first approach where we use a mutable boolean.